main.tf

```diff
@@ -132,7 +132,7 @@ module "cloudwatch_log" {
 resource "aws_iam_saml_provider" "default" {
   count = var.saml_metadata_document != null ? 1 : 0
 
-  name                   = var.name
+  name                   = module.this.id
```

Comment:
Would changing the name cause a recreation of this resource ?

Reply:
Yes. Changing the name of the resource would result in a creation.